hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/headers/Http2RequestHeadersProvider.java

```diff
@@ -0,0 +1,43 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http.headers;
+
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.jetty.http.HttpHeader;
+import pl.allegro.tech.hermes.api.ContentType;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.auth.HttpAuthorizationProvider;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static pl.allegro.tech.hermes.api.AvroMediaType.AVRO_BINARY;
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+
+public final class Http2RequestHeadersProvider implements HttpRequestHeadersProvider {
+
+    private static final Function<ContentType, String> contentTypeToMediaType = contentType ->
+            AVRO.equals(contentType) ? AVRO_BINARY : APPLICATION_JSON;
+
+    private final Optional<HttpAuthorizationProvider> authorizationProvider;
+
+    public Http2RequestHeadersProvider(Optional<HttpAuthorizationProvider> authorizationProvider) {
```

Comment:
Same as in `Http1RequestHeadersProvider`, please no Optional as argument. :)

Reply:
Sure.